auth-foundation/src/main/java/com/okta/authfoundation/credential/Credential.kt

```diff
@@ -106,6 +105,85 @@ class Credential internal constructor(
         }
     }
 
+    companion object {
+        private val defaultCredentialIdDataStore: DefaultCredentialIdDataStore
+            get() = DefaultCredentialIdDataStore.instance
+
+        @VisibleForTesting
+        internal suspend fun credentialDataSource() = CredentialDataSource.getInstance()
+
+        /**
+         * Returns the default [Credential], or null if no default [Credential] exists.
+         */
+        suspend fun getDefaultCredential(): Credential? {
+            return defaultCredentialIdDataStore.getDefaultCredentialId()?.let { id -> with(id) }
+        }
+
+        /**
+         * Sets the default [Credential] to provided [credential].
+         */
+        suspend fun setDefaultCredential(credential: Credential) {
+            defaultCredentialIdDataStore.setDefaultCredentialId(credential.id)
+            AuthFoundationDefaults.eventCoordinator.sendEvent(DefaultCredentialChangedEvent(credential))
+        }
+
+        /**
+         * Returns IDs of all available [Credential] objects in storage.
+         */
+        suspend fun allIds(): List<String> = credentialDataSource().allIds()
```

Comment:
Likewise, instead of a function, is it possible for this to be a `var` with a custom getter?